File: Venda-distribuida-java/src/main/java/br/ufc/venda/client/dao/VendaDAO.java
package br.ufc.venda.client.dao;

import java.sql.Connection;
import java.sql.Date;
import java.sql.PreparedStatement;
import java.sql.ResultSet;

import br.ufc.venda.model.Venda;
import br.ufc.venda.client.connection.ConnectionFactoryReplica;

public class VendaDAO {
	
	private Connection connection;
	private PreparedStatement stmt;
	private ResultSet rs;
	
	public boolean salvar(Venda venda){
		try{
			connection = ConnectionFactoryReplica.getConnection();
			stmt = connection.prepareStatement("INSERT INTO venda (cliente_id, produto_id, quantidade, total, data_venda, numero_cartao, pagamento) VALUES (?,?,?,?,?,?,?)");
			stmt.setLong(1, venda.getCliente().getCpf());
			stmt.setLong(2, venda.getProduto().getId());
			stmt.setInt(3, venda.getQuantidade());
			stmt.setDouble(4, venda.getTotal());
			
			Date sqlDate = new Date(venda.getDataVenda().getTime()); 
			stmt.setDate(5, sqlDate);
			stmt.setLong(6, venda.getPagamento().getNumCartao());
			stmt.setInt(7, venda.getPagamento().getModoPagamento().pagamento());
			
			stmt.execute();
			stmt.close();
			
			return true;
		}catch (Exception e) {
			System.out.println("Não foi possível salvar a venda: "+e.getMessage());
			e.printStackTrace();
		}
		return false;
	}
	
	public boolean excluir(Venda venda){
		try{
			connection = ConnectionFactoryReplica.getConnection();
			stmt = connection.prepareStatement("DELETE FROM venda WHERE cliente_id = ? AND produto_id = ? AND quantidade = ? AND total = ? AND data_venda = ?");
			stmt.setLong(1, venda.getCliente().getCpf());
			stmt.setLong(2, venda.getProduto().getId());
			stmt.setInt(3, venda.getQuantidade());
			stmt.setDouble(4, venda.getTotal());
			
			Date sqlDate = new Date(venda.getDataVenda().getTime()); 
			stmt.setDate(5, sqlDate);
			
			stmt.execute();
			stmt.close();
			
			return true;
		}catch (Exception e) {
			System.out.println("Não foi possível excluir a venda: "+e.getMessage());
			e.printStackTrace();
		}
		return false;
	}
	
}

File: Venda-distribuida-python/dao/VendaDAO.py
#-*- coding:utf-8 -*-

from connection.ConnectionFactory import ConnectionFactory

class VendaDAO():

    def salvar(self, venda):
        try:
            con = ConnectionFactory.getConnection().cursor()
            con.execute('INSERT INTO venda (cliente_id, produto_id, quantidade, total, data_venda, numero_cartao, pagamento) VALUES ('+venda['cliente']['id']+','+venda['produto']['id']+','+venda['quantidade']+','+venda['total']+','+venda['dataVenda']+','+venda['numCartao']+''+venda['modoPagamento']+')')
            return True
        except Exception as e:
            print('Não foi possível salvar a venda:' + str(e))
        return False

    def excluir(self, venda):
        try:
            con = ConnectionFactory.getConnection().cursor()
            con.execute('DELETE FROM venda WHERE cliente_id = '+venda['cliente']['id']+' AND produto_id = '+venda['produto']['id']+' AND quantidade = '+venda['quantidade']+' AND total = '+venda['total']+' AND data_venda = '+venda['dataVenda'])
            return True
        except Exception as e:
            print('Nao foi possivel excluir a venda: '+str(e))
        return False